Motion/app/src/main/java/com/example/android/motion/demo/containertransform/CheeseArticleFragment.kt

```diff
@@ -111,5 +118,45 @@ class CheeseArticleFragment : Fragment() {
         toolbar.setNavigationOnClickListener { v ->
             v.findNavController().popBackStack()
         }
+
+        requireActivity().onBackPressedDispatcher.addCallback(
+            viewLifecycleOwner,
+            object : OnBackPressedCallback(true) {
+                override fun handleOnBackPressed() {
+                    // This invokes the sharedElementReturnTransition, which is
+                    // MaterialContainerTransform.
+                    // TODO: It pops the scaleX/Y back to 1f. It would look nicer to retain the
+                    //       scale. How should we handle it?
+                    findNavController().popBackStack()
+                }
+
+                override fun handleOnBackProgressed(backEvent: BackEventCompat) {
+                    val progress = backEvent.progress
+                    // Translate as far as 20% of finger movement.
+                    val translation = progress * background.width * 0.2f
+                    background.translationX =
+                        if (backEvent.swipeEdge == BackEventCompat.EDGE_LEFT) {
+                            translation
+                        } else {
+                            -translation
+                        }
+                    // TODO: Consider handling backEvent.touchY to reflect the vertical movement.
+
+                    // Scale down from 100% to 50%.
+                    val scale = (2f - progress) * 0.5f
```

Comment:
the formula is 1F - (0.1F * backEvent.progress) because the max scale is 90%

Reply:
Updated. I suppose you mean the "min" scale.